### PR TITLE
style: simplify string formatting for readability

### PR DESCRIPTION
**Pull request by hamirmahal**
**Description:**

This change cleans up string formatting to improve code readability and maintainability.

There are other changes, too, like removing `mut` where we don't need it.

### CODE REVIEW

File: crates/compiler/src/update/patch_module_group_graph/tests.rs [hunk 1]
```diff
@@ -214,9 +212,7 @@ fn test_patch_module_group_graph_3() {
   let update_module_group_graph = module_group_graph_from_entries(
     &module_graph
       .entries
-      .clone()
-      .into_iter()
-      .map(|(entry, _)| entry)
```
Comment (hamirmahal) [inline crates/compiler/src/update/patch_module_group_graph/tests.rs:219]: This is a pattern that shows up a lot. This pull request replaces it with

```rust
.clone().into_keys()
```

Comment (wre232114) [reply crates/compiler/src/update/patch_module_group_graph/tests.rs:219]: thanks

**Review by wre232114 — APPROVED** [review]

**Comment by wre232114** [comment]
Thanks!

**Comment by hamirmahal** [comment]
You're welcome!

**Pull request merged**